### ISSUE TITLE
Connection reset by peer

### ISSUE THREAD

**Issue by khenidak** [Azure/azure-sdk-for-go#14488]
### Bug Report

- import path of package in question, e.g. `.../services/compute/mgmt/2018-06-01/compute`
**github.com/Azure/azure-sdk-for-go/storage**
- SDK version e.g. `master`, `latest`, `18.1.0`
**v50.0.0+incompatible**
-Specify the exact commit if possible; one way to get this is the REVISION
**v50.0.0+incompatible**
    column output by `dep status "github.com/Azure/azure-sdk-for-go`.
- output of `go version`
**go version go1.15 linux/amd64**

- What happened?
We are seeing too many `connection reset by peer..` errors when connecting to Azure Table storage. Client is not behind firewalls and has a public PIP (it is running on Azure VM in the same region as the storage account).

- What did you expect or want to happen?
There are two components to this:
1. We should not be seeing that many (right now we have a few every 2/3 mins test runs).
2. The SDK should retry these errors. Specially when they are not DNS or `connect()` errors.

- How can we reproduce it?
```
git clone https://github.com/khenidak/london/
cd london
make integration-tests # you have to create testing vars in ./hack/testing_vars file
```

**Comment by ArcturusZhang** [comment]
@khenidak thanks for this issue!

@jhendrixMSFT could you please take a look at this storage SDK issue?

**Comment by jhendrixMSFT** [comment]
The fix has been merged, assigning to @ArcturusZhang to track for the next release.

**Comment by jhendrixMSFT** [comment]
Fixed in v54.0.0
